Code/Framework/AzToolsFramework/Tests/PythonLoaderTests.cpp

```diff
@@ -0,0 +1,126 @@
+
+        AZStd::string engineRoot{ s_testEnginePath };
+
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(test3rdPartyPath.String().c_str());
+
+        auto result = AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonVenvPath(test3rdPartyPath, engineRoot);
+
+        AZ::IO::FixedMaxPath expectedPath = testVenvPath / s_testEnginePathHashId;
+
+        EXPECT_TRUE(result == expectedPath);
+    }
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestGetPythonVenvExecutablePath_Valid)
+    {
+        // Prepare the test venv pyvenv.cfg file in the expected location
+        AZ::IO::FixedMaxPath tempVenvRelativePath = AZ::IO::FixedMaxPath(s_test3rdPartySubPath) / "venv/" / s_testEnginePathHashId;
+        AZ::IO::FixedMaxPath tempVenvFullPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / tempVenvRelativePath;
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(tempVenvFullPath.String().c_str());
+        AZ::IO::FixedMaxPath tempPyConfigFile = tempVenvRelativePath / "pyvenv.cfg";
+        AZStd::string testPython3rdPartyPath = "/home/user/python/";
+        AZStd::string testPyConfigFileContent = AZStd::string::format("home = %s\n"
+                                                                      "include-system-site-packages = false\n"
+                                                                      "version = 3.10.13",
+                                                                      testPython3rdPartyPath.c_str());
+        AZ::Test::CreateTestFile(m_tempDirectory, tempPyConfigFile, testPyConfigFileContent);
+
+
+        // Test the method
+        AZStd::string engineRoot{ s_testEnginePath };
+        AZ::IO::FixedMaxPath test3rdPartyRoot = m_tempDirectory.GetDirectoryAsFixedMaxPath() / s_test3rdPartySubPath;
+
+        auto result = AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonExecutablePath(test3rdPartyRoot, engineRoot);
+        AZ::IO::FixedMaxPath expectedPath{ testPython3rdPartyPath };
+
+        EXPECT_TRUE(result == expectedPath);
+    }
+
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestReadPythonEggLinkPaths_Valid)
+    {
+        // Prepare the test folder and create dummy egg-link files
+        AZ::IO::FixedMaxPath testRelativeSiteLIbsPath = AZ::IO::FixedMaxPath(s_test3rdPartySubPath) / "venv" / s_testEnginePathHashId / LY_PYTHON_SITE_PACKAGE_SUBPATH;
+        AZ::IO::FixedMaxPath testFullSiteLIbsPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / testRelativeSiteLIbsPath;
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(testFullSiteLIbsPath.String().c_str());
+
+        AZStd::vector<AZStd::string> expectedResults;
+        expectedResults.emplace_back(testFullSiteLIbsPath.LexicallyNormal().Native());
+
+        static constexpr const char* testEggLinkPaths[] = { "/lib/path/one", "/lib/path/two", "/lib/path/three" };
+        size_t testEggLinkPathCount = sizeof(testEggLinkPaths) / sizeof(const char*);
+
+
+        for (size_t i = 0; i < testEggLinkPathCount; ++i)
```

Comment:
You can use [AZStd::to_array](https://en.cppreference.com/w/cpp/container/array/to_array) here to initialize an `AZStd::array` without needing to explicitly specify the size.
```suggestion
        static constexpr auto testEggLinkPaths = AZStd::to_array<const char*>({ "/lib/path/one", "/lib/path/two", "/lib/path/three" });
        
        for (const char* testEggLinkPath : testEggLinkPaths)
```